### PR TITLE
feat: support whole config as env

### PR DESCRIPTION
**Pull request by eliobischof**
Follow-up for #6143

- Uses custom decoders for complex data structures, so all config properties can be configured via env variables
- Fixes some wrong env var descriptions

### Definition of Ready

- [x] I am happy with the code
- [x] Short description of the feature/issue is added in the pr description
- [ ] PR is linked to the corresponding user story
- [x] Acceptance criteria are met
- [x] All open todos and follow ups are defined in a new ticket and justified
- [x] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [ ] No debug or dead code
- [x] My code has no repetitions
- [x] Critical parts are tested automatically
- [ ] Where possible E2E tests are implemented
- [x] Documentation/examples are up-to-date
- [x] All non-functional requirements are met
- [ ] Functionality of the acceptance criteria is checked manually on the dev system.

### CODE REVIEW

File: cmd/defaults.yaml [hunk 1]
```diff
@@ -318,6 +320,7 @@ Console:
     SharedMaxAge: 168h # ZITADEL_CONSOLE_LONGCACHE_SHAREDMAXAGE
   InstanceManagementURL: "" # ZITADEL_CONSOLE_INSTANCEMANAGEMENTURL
 
+# TODO: Is this still used?
```
Comment (eliobischof) [inline cmd/defaults.yaml:323]: @adlerhurst can we remove this section?

**Comment by eliobischof** [comment]
@adlerhurst I resolved the conflicts

File: cmd/start/config.go [hunk 2]
```diff
@@ -112,3 +121,31 @@ func MustNewConfig(v *viper.Viper) *Config {
 
 	return config
 }
+
+func sliceTypeEnvDecodeHookFunc[T any](from, to reflect.Value) (any, error) {
```
Comment (adlerhurst) [inline cmd/start/config.go:125]: I'd prefer to move the hooks to a separate package so that all cli commands can use them

**Review by adlerhurst — CHANGES_REQUESTED** [review]

File: cmd/start/config.go (hunk 2 above)
Comment (eliobischof) [reply cmd/start/config.go]: I moved them to a new hooks package.
Also, I removed the specific system users hook and use the generic map hook instead. Probably, the system users hook came in when I updated the branch and I overlooked it.

**Review by adlerhurst — APPROVED** [review]

**Pull request merged**